### ISSUE TITLE
Better error message for dealing with bad constructors

### ISSUE THREAD

**Issue by rahulmutt** [yesodweb/persistent#537]
config/models
<pre>
Test
   field Text.
</pre>

Error:
<pre>
Model.hs:13:7:
    Illegal type constructor or class name: ‘Text.’
</pre>
and it goes on with a huge dump of TH splices.

In practice, it's unlikely any user will actually face this unless they accidentally typo, but it's good practice to give good error messages.

The fix is as follows: 
In persistent, Database.Persist.Quasi, Lines 67-69,
```haskell
    getCon t =
        case T.breakOnEnd "." t of
            (_, "") -> FTTypeCon Nothing t
```
should be
```haskell
    getCon t =
        case T.breakOnEnd "." t of
            (t', "") -> error ("Invalid data constructor - dangling qualifier: " ++ T.unpack t')
```

Since this is harmless, maybe it can be snuck in a code refactor commit in the future?

**Comment by gregwebs** [comment]
Did you want to send this as a PR? It would be best to check for any character that is illegal in a type name.

**Comment by rahulmutt** [comment]
Yeah I did, but I was busy at the time and I didn't get the chance. I'll work on it now. I think with an illegal character in a type, the error will just be "type not found" which is fine, but I'll play around with different illegal combinations and see what happens and fix it accordingly.

**Comment by parsonsmatt** [comment]
This is a great newcomer ticket because the implementation is already done ;) Thanks @rahulmutt !
